primitives/session/src/lib.rs

```diff
@@ -106,6 +108,69 @@ impl GetValidatorCount for MembershipProof {
 	}
 }
 
+/// A trait for online node inspection in a session.
+///
+/// This trait is used to decouple the pallet-session dependency from im-online
+/// module so that the user of im-online & offences modules can pass any list of
+/// validators that are considered to be online in each session, particularly useful
+/// for the Substrate-based projects having their own staking implementation
+/// instead of using pallet-staking directly.
+pub trait ValidatorSet<AccountId> {
+	// TODO [ToDr] This could use `frame_support::Parameter` instead,although don't know if such
+	// import is legal.
+	type ValidatorId: codec::Codec + codec::EncodeLike + Clone + Eq + sp_std::fmt::Debug;
+	// TODO [ToDr] This is most likely not needed along with `AccountId`
+	type ValidatorIdOf: Convert<AccountId, Option<Self::ValidatorId>>;
```

Comment:
@tomusdrw Not sure what your final idea about the TODOs here. Use geneic?

```rust
pub trait ValidatorSet<AccountId, ValidatorId> {
          type ValidatorIdOf: Convert<AccountId, Option<ValidatorId>>;
```

Reply:
Since afaict the conversion is not used at all, I meant to remove it completely along with the `AccountId` generic parameter.
```rust
pub trait ValidatorSet {
  type ValidatorId: ...;
  ...
}
```

Reply:
`ValidatorIdOf` is used in https://github.com/paritytech/substrate/pull/7127/files#diff-7cfdd07d0dc9e727202ed41434c4f0f8caaa0329e2abfe717dc673b0794367f5R195, I guess we'll keep it then.